### DISCUSSION TITLE
Passing props to Layout

### DISCUSSION THREAD

**Discussion by AdaptingAFM** [vercel/next.js#33346] [Help] [answered]
I was playing around and making a Dashboard and so I decided to share layout between the multiple pages my dashboard has. I need to pass some props to my layout, but I can't get my way to passing any prop from the page to the getLayout function. 

Goal is to pass to DashboardLayout several props whose's state and data come from the page itself: `loading` - which is provided to the page by an auth middleware; `session` - an object with the session data; `store` - the returning value from swr.

```js

const Dashboard = ({loading}) => {
	const { data: session } = useSession();

	const { data, error, mutate } = useSWR(
		loading ? null : `/api/stores/${session.user.store}?type=id`,
		get
	);

	return <>a</>;
};

Dashboard.auth = {
	loading: <Dashboard loading={true} />,
};

Dashboard.getLayout = function getLayout(page: any) {

	return <DashboardLayout theme={theme}>{page}</DashboardLayout>;
};
```

**Comment by AdaptingAFM** [comment, marked as answer]
I solved my issue today. Will maybe make a repo with the solution in a few days from now.

**Reply by qhkm** [reply]
I'm trying to achieve the same thing. Could you please share your solution @AdaptingAFM. Thank you very much!

**Reply by AdaptingAFM** [reply]
At the end, I was not required of passing the props from the page to the layout, but just to leverage that data to a ContextProvider within the layout itself (and doing the required modifications at the _app component for it to run the getLayout function inside the contexts I was in need of - that is, authentication providers, themes, etc.)

If you need to pass some static or serverSideProps from the page to the component, you could clone the page's layout (with reactjs functions) and pass the pageProps that your _app gets to it.

Remember that you can check the module exported (that is, your page component) name, and so you can modify the behavior of _app in specific cases.

**Reply by daxidngyn** [reply]
Hey there! I'm trying to achieve this same thing right now and was wondering if you ever got around to publishing the repo with the solution?

Would help a lot, much appreciated!

**Reply by AdaptingAFM** [reply]
> Hey there! I'm trying to achieve this same thing right now and was wondering if you ever got around to publishing the repo with the solution?
> 
> Would help a lot, much appreciated!

@daxidngyn Sadly I do not toy around anymore with programming - and so I lack the motivation to publish a repo of my own -, though if you do need help I wouldn't mind making an implementation to your domain or to any repo you point me towards.
